crates/ra_proc_macro_srv/src/dylib.rs

```diff
@@ -123,9 +123,12 @@ impl Expander {
             .canonicalize()
             .unwrap_or_else(|err| panic!("Cannot canonicalize {}: {:?}", lib.display(), err));
 
+        // Copy the dylib to temp directory to prevent locking in Windows
+        let lib = copy_to_temp_dir(&lib).map_err(|e| e.to_string())?;
+
         let library = ProcMacroLibraryImpl::open(&lib).map_err(|e| e.to_string())?;
```

Comment:
I wonder if coverting errors to string is the right approach. I might be mistaken, but if we change this to
```rust
format!({:?}, err)
```
we will be able to get extra information, but I am not sure. E.g. the error message I got ([`Dynamic loading not supported`](https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Fwg-rls-2.2E0/topic/proc-macro/near/195287524)) and the error backtrace don't indicate that it came exactly from musl implementation of `dlopen()`.
Also the approach with bare `String` as error suffers from the inability to convey the error backtrace.

Reply:
I changed the error handling here to just propagate to the caller of this function. However, it will still convert the error to `String` for serialization to the client side.